pytorch_lightning/utilities/device_parser.py

```diff
@@ -199,15 +219,37 @@ def _normalize_parse_gpu_input_to_list(gpus: Union[int, List[int], Tuple[int, ..
     if not gpus:  # gpus==0
         return None
     if gpus == -1:
-        return _get_all_available_gpus()
+        return _get_all_available_gpus(include_cuda=include_cuda, include_mps=include_mps)
 
     return list(range(gpus))
 
 
-def _get_all_available_gpus() -> List[int]:
+def _get_all_available_gpus(include_cuda: bool = False, include_mps: bool = False) -> List[int]:
+    """
+    Returns:
+        a list of all available gpus
+    """
+    cuda_gpus = _get_all_available_cuda_gpus() if include_cuda else []
+    mps_gpus = _get_all_available_mps_gpus() if include_mps else []
+    # This is where in the future ROCM parsing for GPUs should also be called
```

Comment:
I don't think we should add this type of comment. So easy for it to become stale

Reply:
will remove it :)